src/ORM/ResultSet.php

```diff
@@ -217,21 +217,23 @@ public function rewind()
      */
     public function valid()
     {
-        if (isset($this->_results[$this->_index])) {
+        if ($this->_index >= $this->_count) {
+            if ($this->_statement !== null) {
+                $this->_statement->closeCursor();
+            }
```

Comment:
They implemented something that makes more sense when you think about it. If you have a SplFIxedArray(5), then the position [4], for example, is definitely set (with null). We could argue that it changed the semantics of `isset` for arrays that return false when the key exists but the value is null, though.